lib/auth/auth.go

```diff
@@ -2541,6 +2562,183 @@ func (a *Server) DeleteNetworkRestrictions(ctx context.Context) error {
 	return a.Services.Restrictions.DeleteNetworkRestrictions(ctx)
 }
 
+// newKeySetForKeySet returns a new CAKeySet with local keys of the same types
+// (SSH, TLS, or JWT) as the argument keySet
+func newKeySetForKeySet(keyStore keystore.KeyStore, clusterName string, keySet types.CAKeySet) (types.CAKeySet, error) {
```

Comment:
this feels a bit awkward
can the keys be picked based on the CA type instead, like in `createSelfSignedCA`?

Reply:
yeah, I've updated this now